### DISCUSSION TITLE
StreamedResponse not working as intended

### DISCUSSION THREAD

**Discussion by servetoz** [symfony/symfony#57413] [Q&A] [answered]
Hi,

I'm developing an application with laravel v11 (uses symfony/http-kernel 7.0), which has some real-time implementations.
For that, I'm using server sent events with symfony's StreamedResponse class.
However, it is not working as intended. It is waiting a callback function to finish then it sends all the data.
If I implement same logic with pure PHP script on the same server, it is working.

I'm very confused because I don't think the problem lies on server configuration (nginx apache2 in plesk) since php script is running with same configurations.
I believe Symfony or Laravel doing something internal that causes this issue. 

Here is the code with StreamedResponse:
```
Route::get('/test', function () {
    @ob_end_clean();
    ini_set('output_buffering', 0);
    ob_implicit_flush(1);
    @ini_set('output_buffering','Off');
    @ini_set('zlib.output_compression',0);
    @ini_set('implicit_flush',1);
    @ob_end_clean();
    set_time_limit(0);

    $response = new StreamedResponse();

    $response->setCallback(function() use ($response) {
        $i = 0;
        ob_start();
		while (1) {
			$time = date('r');
            // ob_start();
			echo "data: The server time is: {$time}\n\n";
			ob_flush();
			flush();
			$i++;
			if ($i > 5) {
				break;
			}
			sleep(1);
		}
    });

    $response->headers->set('Content-Type', 'text/event-stream');
    $response->headers->set('X-Accel-Buffering', 'no');
    $response->headers->set('Cache-Control', 'no-cache');
    $response->headers->set('Connection', 'keep-alive');


    return $response;
});
```

And result:

https://github.com/symfony/symfony/assets/46219628/dd71f98a-5d8b-4892-abf7-106e6ef584fb

And this is the same logic with PHP script:
```
<?php

header('Content-Type: text/event-stream');
header('Cache-Control: no-cache, private');
//header('Access-Control-Allow-Credentials: true');
//header('Access-Control-Allow-Methods: GET');
//header('Access-Control-Expose-Headers: X-Events');
//header('Access-Control-Allow-Origin: *');
header('X-Accel-Buffering: no');
header('Connection: keep-alive');

$i = 0;
while (1) {
    $time = date('r');
    echo "data: The server time is: {$time}\n\n";
    ob_flush();
    flush();
    $i++;
    if ($i > 5) {
        die();
    }
    sleep(1);
}
?>
```

https://github.com/symfony/symfony/assets/46219628/22f6c642-69dc-46bf-94f7-5adedad1f243

**Comment by MatTheCat** [comment]
Hello,

your PHP configuration probably has `output_buffering` enabled. Calling `ob_start` in the `StreamedResponse` callback will open another buffer (see https://www.php.net/manual/en/outcontrol.nesting-output-buffers.php) so when you call `ob_flush`, only that buffer is flushed and its content ends up in the parent buffer.

TL;DR removing the call to `ob_start` in the `StreamedResponse` callback should do the job.

**Comment by servetoz** [comment, marked as answer]
Hi,

The issue was because of weird configuration of plesk. Plesk uses combination nginx and apache to handle websites.

This is nginx.conf:
```
 location / {
                proxy_read_timeout 120;
                proxy_pass https://IP:7081;
                proxy_hide_header upgrade;
                proxy_set_header Host             $host;
                proxy_set_header X-Real-IP        $remote_addr;
                proxy_set_header X-Forwarded-For  $proxy_add_x_forwarded_for;
                proxy_set_header X-Accel-Internal /internal-nginx-static-location;
                access_log off;

        }
 location ~ \.php(/.*)?$ {
                fastcgi_read_timeout 120;
                fastcgi_split_path_info ^((?U).+\.php)(/?.+)$;
                try_files $uri $fastcgi_script_name =404;
                fastcgi_param PATH_INFO $fastcgi_path_info;
                fastcgi_pass "unix:/var/www/vhosts/system/site/php-fpm.sock";
                include /etc/nginx/fastcgi.conf;
        }
```
So when I run php script nginx passes the request directly to fastcgi, however all requests "/" goes through apache then fastcgi.
The problem was actually in the configuration.

Adding following line configuration in apache httpd.conf fixed the issue:
```
<Proxy "fcgi://127.0.0.1:9000">
	ProxySet enablereuse=on
	ProxySet flushpackets=on
</Proxy>
```
